src/parse.ts

```diff
@@ -1,109 +1,109 @@
   disableDerivesFromReferences?: boolean
 }
 
 interface References {
-  Parent: GFF3.GFF3Feature[]
-  Derives_from: GFF3.GFF3Feature[]
+  Parent: GFF3Feature[]
+  Derives_from: GFF3Feature[]
 }
 
-export default class Parser {
-  featureCallback: (feature: GFF3.GFF3Feature) => void
+export class GFF3Parser {
   endCallback: () => void
-  commentCallback: (comment: GFF3.GFF3Comment) => void
-  errorCallback: (error: string) => void
   disableDerivesFromReferences: boolean
-  directiveCallback: (directive: GFF3.GFF3Directive) => void
-  sequenceCallback: (sequence: GFF3.GFF3Sequence) => void
   bufferSize: number
   fastaParser: FASTAParser | undefined = undefined
-  // if this is true, the parser ignores the
-  // rest of the lines in the file.  currently
-  // set when the file switches over to FASTA
+  // if this is true, the parser ignores the  rest of the lines in the file.
+  // currently set when the file switches over to FASTA
   eof = false
   lineNumber = 0
-  // features that we have to keep on hand for now because they
-  // might be referenced by something else
-  private _underConstructionTopLevel: GFF3.GFF3Feature[] = []
+  // features that we have to keep on hand for now because they might be
+  // referenced by something else
+  private underConstructionTopLevel: GFF3Feature[] = []
   // index of the above by ID
-  private _underConstructionById: Record<string, GFF3.GFF3Feature | undefined> =
-    {}
-  private _completedReferences: Record<
+  private underConstructionById: Record<string, GFF3Feature | undefined> = {}
+  private completedReferences: Record<
     string,
     Record<string, boolean | undefined> | undefined
   > = {}
-  // features that reference something we have not seen yet
-  // structured as:
+  // features that reference something we have not seen yet. structured as:
   // {  'some_id' : {
   //     'Parent' : [ orphans that have a Parent attr referencing it ],
   //     'Derives_from' : [ orphans that have a Derives_from attr referencing it ],
   //    }
   // }
-  private _underConstructionOrphans: Record<string, References | undefined> = {}
+  private underConstructionOrphans: Map<string, References | undefined> =
+    new Map()
 
   constructor(args: ParserArgs) {
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     const nullFunc = () => {}
 
-    this.featureCallback = args.featureCallback || nullFunc
     this.endCallback = args.endCallback || nullFunc
-    this.commentCallback = args.commentCallback || nullFunc
-    this.errorCallback = args.errorCallback || nullFunc
-    this.directiveCallback = args.directiveCallback || nullFunc
-    this.sequenceCallback = args.sequenceCallback || nullFunc
     this.disableDerivesFromReferences =
       args.disableDerivesFromReferences || false
 
     // number of lines to buffer
-    this.bufferSize = args.bufferSize === undefined ? 1000 : args.bufferSize
+    this.bufferSize = args.bufferSize === undefined ? 50000 : args.bufferSize
```

Comment:
I do want to call out this change, I have run into buffer size being not enough and it is a maddening error. I think it is actually a potential source of serious mysterious bugs and it may be worth removing it entirely, making it infinity (similar to chunk size error or similar...just something the user shouldnt have to configure)